From: Richie Roy Jayme <rjayme.jp@gmail.com>
Date: Wed, 25 Mar 2026 18:33:21 +0900
Subject: [PATCH] quirks: add support for Goodix touchpad GXTP5400:00
 27C6:0F96

This commit adds a specific vendor HWID for Goodix Haptic Touchpad to
improve detection and handling.

Signed-off-by: Richie Roy Jayme <rjayme.jp@gmail.com>
Signed-off-by: Richie Roy Jayme <rjayme2@lenovo.com>
Reviewed-by: Vishnu Sankar <vishnuocv@gmail.com>
Reviewed-by: Vishnu Sankar <vsankar@lenovo.com>
Tested-by: Ameer Ivan Julkarnain <ajulkarnain1@lenovo.com>
Part-of: <https://gitlab.freedesktop.org/libinput/libinput/-/merge_requests/1455>

Origin: backport, https://gitlab.freedesktop.org/libinput/libinput/-/commit/baddf1e2b6db43d4e726d58a940f7720d9368c57
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2146518
Signed-off-by: Bin Li <binli@ubuntu.com>
---
 quirks/30-vendor-goodix.quirks | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/quirks/30-vendor-goodix.quirks b/quirks/30-vendor-goodix.quirks
index e294156..53b0391 100644
--- a/quirks/30-vendor-goodix.quirks
+++ b/quirks/30-vendor-goodix.quirks
@@ -9,6 +9,15 @@ MatchProduct=0x01E8
 MatchUdevType=touchpad
 AttrInputProp=+INPUT_PROP_PRESSUREPAD
 
+# "GXTP5400 Touchpad": pressure touchpad mostly used in Lenovo laptops.
+# GXTP5400:00 27C6:0F96 Touchpad
+[Goodix Haptic Touchpad (27C6:0F96)]
+MatchBus=i2c
+MatchVendor=0x27C6
+MatchProduct=0x0F96
+MatchUdevType=touchpad
+AttrInputProp=+INPUT_PROP_PRESSUREPAD
+
 # "GXTP5420 Touchpad": pressure touchpad mostly used in Lenovo laptops.
 # GXTP5420:00 27C6:0F95 Touchpad
 [Goodix Haptic Touchpad (27C6:0F95)]
